### PR TITLE
Update Ogre to 1.10.11

### LINKED ISSUE #170: Update Ogre and change billboard_chain
The `billboard_chain.cpp` shouldn't need to count the number of lines per chain since chains can do this on their own. However, the functionality is broken in the version we use (here is the issue: https://github.com/OGRECave/ogre/issues/603). 

- Update the Ogre version at some point.
- Remove num_elements in `billboard_chain.cpp`
- Rename `newLine()` to `finishLine()`

### LINKED ISSUE #180: RViz segfault on macOS (v. 10.13.2) with latest master
Following the change from static to dynamic linking, on Apple (on Linux and Windows this problem doesn't occur) RViz crashes at startup with a segfault. 
The problem was traced back to the `params["externalWindowHandle"] = Ogre::StringConverter::toString(window_id);` line, in render_system.cpp (line 428). When it is commented out, the segfault doesn't happen, but, of course, RViz starts with the render window detached from the application window. 

Proposed Solution: apparently, updating to the latest version of Ogre fixes the problem.

### PR DESCRIPTION
**Pull request by Martin-Idel-SI**
Fixes #170 

Update Ogre to Version 1.10.11.
We did some manual testing on Linux, Mac and Windows (VS2015) and it seems everything works as it should.

### CODE REVIEW

**Comment by Martin-Idel-SI** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/3899/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1017/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3222/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/3992/)

File: rviz_ogre_vendor/rviz_ogre_vendor-extras.cmake.in [hunk 1]
```diff
@@ -48,7 +48,7 @@ foreach(_lib IN LISTS OGRE_LIBRARIES)
       NO_CMAKE_SYSTEM_PATH
       NO_CMAKE_FIND_ROOT_PATH
     )
-    if(NOT _ogre_main_static_library_debug_abs AND APPLE)
+    if(NOT _ogre_main_static_library_debug_abs AND NOT WIN32)
```
Comment (wjwwood) [inline rviz_ogre_vendor/rviz_ogre_vendor-extras.cmake.in:51]: Is it not append on Linux too?

**Review by wjwwood — APPROVED** [review]
Looks reasonable, it might need to be rebased and rerun CI to make sure it's still green (since we don't have auto rebasing on forks).

File: rviz_ogre_vendor/rviz_ogre_vendor-extras.cmake.in (hunk 1 above)
Comment (greimela-si) [reply rviz_ogre_vendor/rviz_ogre_vendor-extras.cmake.in:51]: No, this changed with the newest release.

Comment (wjwwood) [reply rviz_ogre_vendor/rviz_ogre_vendor-extras.cmake.in:51]: Ok, can you guys rebase this and rerun the CI? I'll have another look tomorrow.

Comment (greimela-si) [reply rviz_ogre_vendor/rviz_ogre_vendor-extras.cmake.in:51]: Will do, thanks for the review!

**Comment by Martin-Idel-SI** [comment]
Rebased and CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/3911/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1030/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3235/) (unrelated to this change)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4005/)

**Please note**: At first, it seemed that mac was working as it should, however that's not the case, the problem described in #180 is not fixed consistently

**Comment by wjwwood** [comment]
Despite the issue in #180, I'll go ahead and merge this for the other benefits.

**Pull request merged**